### ISSUE TITLE
[chromecast] Player state not updated when changed outside openHAB

### ISSUE THREAD

**Issue by lolodomo** [openhab/openhab2-addons#1783]
I tried the Chromecast binding. It discovered my Freebox Player Mini 4k (running Android TV).
I can use play and pause commands from the control channel.
When I play or pause a YouTube video from my phone using YouTube application, the state of the channel is not updated accordingly.

**Comment by lolodomo** [comment]
spontaneousEventReceived method is never called when you play/pause YouTube outside openHAB.

At the reverse, when I play/pause with openHAB, I can see that YouTube state is updated accordingly in my tablet. So it is probably possible to do the reverse.

**Comment by lolodomo** [comment]
I opened an issue in the library we rely on: https://github.com/vitalidze/chromecast-java-api-v2/issues/63

**Comment by holmes** [comment]
This should be resolved now that https://github.com/openhab/openhab2-addons/pull/2062 is merged.

**Comment by lolodomo** [comment]
@holmes : I just installed in my production environment the snapshot 849. Do you know if PR 2062 is included in this version ?

**Comment by holmes** [comment]
I don't know how to correlate snapshots to SHA. 

FWIW, it was merged last night and I did an update from snapshots this afternoon and it's definitely in that version.

**Comment by lolodomo** [comment]
Ok will try that soon.. Thanks for the enhancements.

**Comment by gersilex** [comment]
It still is not working. But I discovered an interesting thing. It works, but only after you have SENT a command to the Player item. From then on, it also receives the updates from the Chromecast. So this looks like there is something like a connection established or the Chromecast won't just broadcast this but multicast to all devices, that have interacted with it.

**Comment by gersilex** [comment]
In https://github.com/vitalidze/chromecast-java-api-v2/issues/63 it was recommended, to try polling the Status regularily (like chromecastplayer does), as it is not fully clear where the problem originates, yet.

Would that be reasonable? I think using both approaches (subscribing and polling) combined would work out well, although I would increase polling time to 5 seconds.

**Comment by holmes** [comment]
One day I'll finish https://github.com/openhab/openhab2-addons/pull/2097 which fixes this bug.

There's a JAR in there you can drop in until it gets merged.

**Comment by gersilex** [comment]
Thanks, I will test this thoroughly and report any issues I find.

**Comment by FredericMa** [comment]
Hi,

It looks like this is working fine now if switching between play and pause for example but if I STOP playing the state remains "PLAY".
Is it possible that the player doesn't support the stop command yet? Or doesn't process it?

Greetings,
Frederic

**Comment by holmes** [comment]
@FredericMa I thought I implemented STOP correctly: https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastCommander.java#L141

Oddly enough, STOP isn't available in the Player command class. So to send stop you've got to send a STOPMOVE type command, which oddly enough is in the Rollershutter command class. 

Does that make any sense?

**Comment by FredericMa** [comment]
Hi @holmes ,

The funny thing is that sending a stop command to the player item does work.
I have this line in my rule and the chromecast does stop playing when the rule is triggered:
`sendCommand(ChromecasaM, STOP)`
The thing is that the item state remains PLAY instead of switching to STOP.
Probably the issue is the fact that the stop command is not implemented in the player item?
But then it is still strange that sending a stop command to the item does work.

Greetings,
Frederic

**Comment by holmes** [comment]
@FredericMa I think I see the bug. When we get a STOP command from OpenHab we tell Chromecast to stop playing. We then update the mediaStatus w/ null. This in turn does this:

```
    public void updateMediaStatus(final MediaStatus mediaStatus) {
        logger.debug("MEDIA_STATUS {}", mediaStatus);

        // In-between songs? It's thinking? It's not doing anything
        if (mediaStatus == null) {
            callback.updateState(CHANNEL_CURRENT_TIME, UNDEF);
            updateMediaInfoStatus(null);
            return;
        }
```

If it didn't early return, the rest of that method updates control state. Need to find the happy medium between updating media state w/ null and sending in a no-longer playing state to the control channel.

Quick question: does it eventually update the state to stopped? After the next refresh?

Not sure when I'll have time to take a look, but we should probably create a new issue for this.

**Comment by FredericMa** [comment]
Hi @holmes,

I checked your question and if I send a STOP command from a rule, the player item remains in the PLAY state. It never goes to STOP, not even after the next refreshes.
In my rule I now put the state to NULL in order for my logic to work correctly:
`postUpdate(ChromecasaM, NULL)`
